File: capacitor-nfc/android/capacitor-nfc/src/main/java/com/adrynov/capacitor/NfcSettings.java
package com.adrynov.capacitor;

public class NfcSettings {

    /**
     * Scan tags that contain NDEF data that cannot be mapped to a MIME type or URI,
     * or if the tag does not contain NDEF data but is of a known tag technology
     */
    private boolean enableTech;

    /**
     * Scan tags that contain an NDEF payload.
     */
    private boolean enableNdef;

    /**
     * Scan NDEF-formattable tags.
     */
    private boolean enableNdefFormattable;

    public boolean techEnabled() {
        return enableTech;
    }

    public void setTechEnabled(boolean enableTech) {
        this.enableTech = enableTech;
    }

    public boolean ndefEnabled() {
        return enableNdef;
    }

    public void setNdefEnabled(boolean enableNdef) {
        this.enableNdef = enableNdef;
    }

    public boolean ndefFormattableEnabled() {
        return enableNdefFormattable;
    }

    public void setNdefFormattableEnabled(boolean enableNdefFormattable) {
        this.enableNdefFormattable = enableNdefFormattable;
    }
}

File: capacitor-nfc/android/src/main/java/com/adrynov/capacitor/nfc/NfcSettings.kt
package com.adrynov.capacitor.nfc

class NfcSettings {

    /**
     * Allows to scan tags that contain NDEF payload.
     * NDEF is a NFC Forum standard (NFC Data Exchange Format)
     */
    var ndefEnabled: Boolean = false

    /**
     * Allows to scan NDEF-formattable tags.
     */
    var ndefFormattable: Boolean = false

    /**
     * Allows to scan tags that contain NDEF data that cannot be mapped to a MIME type or URI,
     * or when the tag does not contain NDEF data but is of a known tag technology e.g. Mifare Classic
     */
    var techEnabled: Boolean = false
}
